test/db/index.js

```diff
@@ -94,6 +96,109 @@ describe('db', function() {
     });
   });
 
+  if (config.get('db.driver') === 'mysql') {
+    describe('purgeExpiredTokens', function () {
+      var clientIdA;
+      var clientIdB;
+      var userId;
+      var email;
+
+
+      function seedTokens (client, userId, email, count, expiresAt) {
+        var accessTokens = [];
+        for (var i = 0; i < count; i++) {
+          accessTokens.push({
+            clientId: buf(client),
+            userId: buf(userId),
+            email: email,
+            scope: [auth.SCOPE_CLIENT_MANAGEMENT],
+            expiresAt: expiresAt
+          });
+        }
+
+        return Promise.each(accessTokens, function (options) {
+          return db.generateAccessToken(options);
+        });
+      }
+
+      // Inserts 2000 access tokens with the following breakdown
+      // ClientIdA - 500 expired, 500 valid
+      // ClientIdB - 500 expired, 500 valid
+      before('setup clients', function(){
+        email = 'asdf@asdf.com';
+        clientIdA = randomString(8);
+        clientIdB = randomString(8);
+        userId = buf(randomString(16));
+
+        return db.registerClient({
+            id: clientIdA,
+            name: 'ClientA',
+            hashedSecret: randomString(32),
+            imageUri: 'https://example.domain/logo',
+            redirectUri: 'https://example.domain/return?foo=bar',
+            trusted: true
+          })
+          .then( function () {
+            return db.registerClient({
+              id: clientIdB,
+              name: 'ClientB',
+              hashedSecret: randomString(32),
+              imageUri: 'https://example.domain/logo',
+              redirectUri: 'https://example.domain/return?foo=bar',
+              trusted: true
+            });
+          });
+      });
+
+      beforeEach('seed with tokens', function () {
+        return db._write('DELETE FROM tokens;')
+          .then( function () {
+            return seedTokens(clientIdA, userId, email, 500);
+          })
+          .then( function () {
+            return seedTokens(clientIdB, userId, email, 500);
+          })
+          .then( function () {
+            return seedTokens(clientIdA, userId, email, 500, new Date(Date.now() - (1000 * 600)));
+          })
+          .then( function () {
+            return seedTokens(clientIdB, userId, email, 500, new Date(Date.now() - (1000 * 600)));
+          });
+      });
+
+      it('should call purgeExpiredTokens', function() {
+        return db.purgeExpiredTokens(1000, 5)
```

Comment:
It's pretty important that, in production, we don't accidentally call this procedure with a `NULL` ignoreClientId.  I wonder if we can make it more robust by e.g. reading the value from config rather than requiring the caller to provide it every time.